### ISSUE TITLE
How to launch with a defined theme?

### ISSUE THREAD

**Issue by echarles** [jupyterlab/jupyterlab#9436]
I wonder how to launch jupyter lab with a defined theme. For now, I enforce a user setting with eg.

```bash
mkdir -p ~/.jupyter/lab/user-settings/\@jupyterlab/apputils-extension && \
  CONF="{ \"theme\": \"JupyterLab Winter\" }" && \
  cat > ~/.jupyter/lab/user-settings/\@jupyterlab/apputils-extension/themes.jupyterlab-settings  <<EOF
${CONF}
EOF
```

Any better way to achieve this?

**Comment by telamonian** [comment]
That's the only way I know how, and I'm one of the folks who wrote the theming frontend. What did you have in mind?

Really, I think that the way you're doing it is "correct", though terrible to interact with via bash. The theme settings file resides at a well-defined path, and internally has an explicitly defined json schema. And so all you're doing is dumping the appropriate json blob at said appropriate path.

I wouldn't be opposed to some kind of `jupyter labsettings` cmd line tool to make this sort of manipulation easier for devs and sysadmins. So for your example, maybe you would end up using a command like:

```bash
jupyter labsettings set "@jupyterlab/apputils-extension/themes"="JupyterLab Winter"
```

But sort of like the story [about belling the cat](http://read.gov/aesop/003.html), someone will need to actually implement this, because it does not currently exist

**Comment by telamonian** [comment]
Of course, you could also just use a python script instead of a bash one. Much nicer (builtin) json support: https://docs.python.org/3/library/json.html. So that would fix part of the problem (though you'd still have to do your own path handling).

You can find nice example of using the `json` pkg in our own codebase, eg in [`coreconfig.py`](https://github.com/jupyterlab/jupyterlab/blob/master/jupyterlab/coreconfig.py).

**Comment by jasongrout** [comment]
A system administrator can override any system default with a new default (which the user can then override in their personal settings, of course). See the `overrides.json` file mentioned in https://jupyterlab.readthedocs.io/en/stable/user/extensions.html?highlight=overrides.json#settings (the example there is exactly your case - setting a system default theme).

**Comment by telamonian** [comment]
Ah. I did not know about `overrides.json`. That sounds handy.

It would still be nice to have some kind of `jupyter labsettings` tool since currently you do still have to interact with the raw settings json yourself, whether in `overrides.json` or in the user settings directly. Something to think about for jlab 4.0, along with the in-lab user settings GUI

**Comment by echarles** [comment]
Thx @telamonian and @jasongrout. Something like `jupuyter lab --settings-ovverides=apputils-extension.theme:"JupyterLab Winter" would be great.

How do we log such feature requests in the jupyterlab roadmap releases. Does the current tag on this issue `type:Enhancement`enough?
